grouper/api/settings.py

```diff
@@ -1,7 +1,25 @@
-from grouper.settings import Settings, settings as base_settings
+from typing import TYPE_CHECKING
 
+from grouper.settings import Settings
 
-settings = Settings.from_settings(
-    base_settings,
-    {"address": None, "debug": False, "num_processes": 1, "port": 8990, "refresh_interval": 60},
-)
+if TYPE_CHECKING:
+    from typing import Optional
+
+
+class ApiSettings(Settings):
+    """Grouper API server settings."""
+
+    def __init__(self):
```

Comment:
As is, each of these Settings classes requires three calls to really set up:
```
settings = XSettings()
settings.update_from_config(filename)
update_global_settings(settings)
```

I'm a bit wary about this as a pattern -- it's really easy to forget that the additional calls are necessary, and tests are the only case I can see where you wouldn't always call all three. Would personally rather have it be a single call, something like `set_global_settings("api", filename)`. Not sure what the best solution would be; just my two cents.

(And same for each of the other Settings classes)

Reply:
Hm. That's definitely a valid point. I think I was writing it that way because I was hopeful it would be possible to drop the third call (and it is already possible in the tests in some places), but that's not really a good argument against adding a class method that does all of those things at the same time and returns the new `Settings` object. I'll do that before merging this.